### ISSUE TITLE
[css-viewport] [css-container-queries] Zoom and container queries

### ISSUE THREAD

**Issue by emilio** [w3c/csswg-drafts#10268]
How should CSS zoom be handled (if at all) in a `@container` query?

A container query like `@container (width > 100px) { }` is rather weird wrt CSS zoom. Should it use pixels:

 * As considered by the page (un-zoomed).
 * As considered by the container.
 * As considered by the element you're styling.

Safari disagrees with Chrome / Firefox on this test-case:

```html
<!doctype html>
<style>
  .container {
    container-type: inline-size;
    width: 100px;
    height: 100px;
  }
  .child {
    background-color: red;
    height: 50px;
    width: 50px;
    @container (width > 120px) {
      background-color: green;
    }
  }
</style>
<div class="container">
  <div class="child"></div>
</div>
<div class="container" style="zoom: 2">
  <div class="child"></div>
</div>
<div class="container" style="zoom: 2">
  <div class="nested" style="zoom: 2">
    <div class="child"></div>
  </div>
</div>
<div class="outer" style="zoom: 2">
  <div class="container">
    <div class="child"></div>
  </div>
</div>
```

Same for container query relative units, I suspect that the best behavior involves some gymnastics with the container and child effective zoom values. Not sure what browsers implement there (Firefox doesn't zoom them at all so they use the coordinate space of the container).

cc @chrishtr @lilles @andruud @nt1m @mirisuzanne @tabatkins

**Comment by lilles** [comment]
I don't think we made a conscious choice here, but the width/height of the container in CSS pixels as seen by the container just follows from the implementation.

**Comment by mirisuzanne** [comment]
I agree, that feels best aligned to how we resolve relative units on the container.

**Comment by svgeesus** [comment]
It is unclear to me what specific spec changes need to be made on css-conditional-5 as a result of this resolution.

**Comment by chrishtr** [comment]
> It is unclear to me what specific spec changes need to be made on css-conditional-5 as a result of this resolution.

I think the spec change needed there is to say that the [principle box](https://www.w3.org/TR/css-display-3/#principal-box)'s bounds are [unscaled](https://drafts.csswg.org/css-viewport/#unscaled).

Also: I added a [testcase](https://wpt.fyi/results/css-viewport/zoom/container-queries.html?label=experimental&label=master&aligned).
